packages/terra-compact-interactive-list/src/utils/keyHandlerUtils.js

```diff
@@ -0,0 +1,310 @@
+    const inputType = element.type;
+    return validTypes.indexOf(inputType) >= 0;
+  }
+
+  return false;
+};
+
+/**
+ * Move focus to next focusable element outside the container element
+ */
+export const moveFocusFromElement = (containerRef, id, moveForward) => {
+  // get focusable elements from the documen, filter out the container and its children
+  const focusableElements = getFocusableElements(document.body).filter(
+    element => (element.id === id || !containerRef.current.contains(element)),
+  );
+
+  // Identify index of the active element in the DOM excluding data grid children
+  const index = focusableElements.indexOf(containerRef.current);
+  if (index > -1) {
+    // Move focus outside data grid
+    const indexOffset = moveForward ? 1 : -1;
+    let newFocusElement;
+
+    if (index + indexOffset < focusableElements.length) {
+      newFocusElement = focusableElements[index + indexOffset];
+    } else if (focusableElements[0].id !== id) {
+      // eslint-disable-next-line prefer-destructuring
+      newFocusElement = focusableElements[0];
+    }
+
+    if (newFocusElement) {
+      newFocusElement.focus();
+    }
+  }
+};
+
+/**
+ * @param {object} focusedCell - a { row, cell } pair, where row and cell are indexes of currently focused semantic row and a cell within that row.
+ * @param {number} numberOfColumns - a number of visual columns.
+ * @param {boolean} flowHorizontally - sematic rows horizontal flow direction
+ * @param {number} rowsLength - a total number of seamntic rows in the list.
+ * @returns - a { row, cell } indexes pair to focus on.
+ */
+export const handleDownKey = (focusedCell, numberOfColumns, flowHorizontally, rowsLength) => {
+  const { row, cell } = focusedCell;
+  if (flowHorizontally) {
+    const visualColumnNumber = row % numberOfColumns;
+    const nextRow = row + numberOfColumns;
+    if (nextRow > rowsLength - 1) {
+      if (visualColumnNumber === numberOfColumns - 1) {
+        // the last semantic row in the last visual column, focus shouldn't go anywhere
+        return focusedCell;
+      }
+      // the last semantic row in current visual column, focus should move to the next visual column
+      return { row: visualColumnNumber + 1, cell };
+    }
+    return { row: nextRow, cell };
+  }
+  // vertical flow
+  const rowsPerColumn = Math.ceil(rowsLength / numberOfColumns);
+  const numberOfPlaceholders = (rowsPerColumn * numberOfColumns) - rowsLength;
+  if ((numberOfPlaceholders === 0 && row + 1 === rowsLength) || (numberOfPlaceholders !== 0 && row + 1 === rowsLength + numberOfPlaceholders - 1)) {
+    // the last semantic row in the last visual column, focus shouldn't go anywhere
+    return focusedCell;
+  }
+  // focus should go to the next semantic row unless the last row
+  return { row: row + 1 < rowsLength ? row + 1 : row, cell };
```

Comment:
You probably could simplify this by using the Math.min function.

Reply:
Not sure I got how to do that. An example will be appreciated.